src/codec.rs

```diff
@@ -1099,13 +1099,31 @@ impl Decode for bool {
 	}
 }
 
+impl Encode for Duration {
+	fn size_hint(&self) -> usize {
+		mem::size_of::<u64>()
+	}
+
+	fn encode(&self) -> Vec<u8> {
+		let millis = self.as_millis() as u64;
+		millis.encode()
+	}
+}
```

Comment:
Encoding of Duration must hold all information about Duration.
Duration is actually containing secs in u64 and nanos in u32 so you should convert to secs and subsecs_nanos to get all information